### DISCUSSION TITLE
Please Make GitHub LFS Usable

### DISCUSSION THREAD

**Discussion by Malix-Labs** [community/community#128695] [Repositories]
Hey team!

I love GitHub and OSS

I want to use GitHub LFS, and even pay for it, but I can't because of some limitations that makes it unusable for me, and would very much like to see it improves

## Context

- > [After you remove files from Git LFS, the Git LFS objects still exist on the remote storage and will continue to count toward your Git LFS storage quota.](https://docs.github.com/en/repositories/working-with-files/managing-large-files/removing-files-from-git-large-file-storage#git-lfs-objects-in-your-repository:~:text=After%20you%20remove%20files%20from%20Git%20LFS%2C%20the%20Git%20LFS%20objects%20still%20exist%20on%20the%20remote%20storage%20and%20will%20continue%20to%20count%20toward%20your%20Git%20LFS%20storage%20quota.)
	>
	> [To remove Git LFS objects from a repository, **delete and recreate the repository**. When you delete a repository, any associated issues, stars, and forks are also deleted.](https://docs.github.com/en/repositories/working-with-files/managing-large-files/removing-files-from-git-large-file-storage#git-lfs-objects-in-your-repository:~:text=To%20remove%20Git%20LFS%20objects%20from%20a%20repository%2C%20delete%20and%20recreate%20the%20repository.%20When%20you%20delete%20a%20repository%2C%20any%20associated%20issues%2C%20stars%2C%20and%20forks%20are%20also%20deleted.)

- **GitHub LFS Limitations**:
	- **Free**
		- **Storage**: 1GB
		- **Bandwidth**: 1GB
	- **For each 5$**
		- **Storage**: 50GB
		- **Bandwidth**: 50GB

- > [Forking and pulling a repository counts against the parent repository's bandwidth usage](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-storage-and-bandwidth-usage#:~:text=Forking%20and%20pulling%20a%20repository%20counts%20against%20the%20parent%20repository%27s%20bandwidth%20usage)

## Rationale

- Deleting a repository is not a solution, it's a very bad workaround
- Manually requesting to delete GitHub support is tedious, not immediate, time consuming, and cannot be automatized for both the end-users and GitHub
- As forking and pulling a repository counts against the parent repository's bandwidth usage, it is dangerously simple to abuse someone and saturate GitHub bandwidth and [even automate it](https://github.com/search?q=version+https%3A%2F%2Fgit-lfs.github.com%2Fspec%2Fv&type=code)
- Storage grows linearly which the number of files and their respective size
	Bandwidth grows linearly with the number of contributors **multiplied by** the number of files and their respective size
- Bandwidth costs is way much cheaper than Storage
- GitLab doesn't have Bandwidth limitations

## Request

- Automatically delete unreferenced GitHub LFS files from GitHub LFS remote storage
- Removal, or at least great expansion of the GitHub LFS Bandwidth limitations (if not removed, consider taking the number of contributors into account)
- Pull of GitHub LFS files from forks and non-members counting toward their respective GitHub LFS Bandwidth usage
- Otherwise, make a documentation about how it would be possible to use Git LFS with another remote storage than the GitHub LFS remote storage

**Comment by osbyrne** [comment]
The first time I'm reading about Git LFS, and it already seems like a weird program. I'd definitely like to be able to prune useless state; or at least use Git LFS as a _large file syncing solution_ rather than a "historical archive of every snapshot of a large file"

**Comment by GuiSousa2910** [comment]
it's okay

**Reply by Malix-Labs** [reply]
?

**Reply by osbyrne** [reply]
One might wonder what Gui meant by "it's okay"

**Comment by Olivier-BigAdventure** [comment]
This is my most important answer as well. How do I fix things with GitHub LFS?
I don't think datapacks are the actual solution and how do I get rid of unreferenced files?
